compactor/periodic_test.go

```diff
@@ -36,31 +36,59 @@ func TestPeriodic(t *testing.T) {
 
 	tb.Run()
 	defer tb.Stop()
-	checkCompactInterval := retentionDuration / time.Duration(periodDivisor)
-	n := periodDivisor
-	// simulate 5 hours worth of intervals.
-	for i := 0; i < n/retentionHours*5; i++ {
+	// simulate 5 hours
+
+	for i := 0; i < 5; i++ {
 		rg.Wait(1)
-		fc.Advance(checkCompactInterval)
+		fc.Advance(time.Hour)
 		// compaction doesn't happen til 2 hours elapses.
-		if i < n {
+		if i < retentionHours {
 			continue
 		}
-		// after 2 hours, compaction happens at every checkCompactInterval.
+		// after 2 hours, compaction happens at every interval.
+		// at i = 3, t.revs = [1(2h-ago,T=0h), 2(1h-ago,T=1h), 3(now,T=2h)] (len=3) (rev starts from 1)
 		a, err := compactable.Wait(1)
 		if err != nil {
 			t.Fatal(err)
 		}
-		expectedRevision := int64(i + 1 - n)
+		expectedRevision := int64(i - 1)
 		if !reflect.DeepEqual(a[0].Params[0], &pb.CompactionRequest{Revision: expectedRevision}) {
 			t.Errorf("compact request = %v, want %v", a[0].Params[0], &pb.CompactionRequest{Revision: expectedRevision})
 		}
 	}
+}
+
+func TestPeriodicMinutes(t *testing.T) {
+	retentionMinutes := 23
+	retentionDuration := time.Duration(retentionMinutes) * time.Minute
+
+	fc := clockwork.NewFakeClock()
+	rg := &fakeRevGetter{testutil.NewRecorderStream(), 0}
+	compactable := &fakeCompactable{testutil.NewRecorderStream()}
+	tb := newPeriodic(fc, retentionDuration, rg, compactable)
+
+	tb.Run()
+	defer tb.Stop()
+
+	// simulate 115 (23 * 5) minutes
+	for i := 0; i < 5; i++ {
+		rg.Wait(1)
+		fc.Advance(retentionDuration)
```

Comment:
we should call `fc.Advance()` one per retry internal so that the compactor itself has time to update it's `revs` array. The above code `fc.Advance(retentionDuration)` only let compactor to get rev once; hence `len(revs)==1` which is not what it should have.

Reply:
see how the old test code work from the link above